### ISSUE TITLE
Wrong FBSDK version fetched with pod install

### ISSUE THREAD

**Issue by AndriyGo** [firebase/FirebaseUI-iOS#578]
Here is a simple Podfile which I created for testing purposes on a clean project:

```
# Uncomment the next line to define a global platform for your project
platform :ios, '9.0'

target 'testSticker' do
  use_frameworks!
  pod 'FirebaseUI', '~> 5.0' 

end
```

When I run `pod install` I see the following output:

```
Installing Bolts (1.9.0)
Installing BoringSSL (10.0.6)
Installing FBSDKCoreKit (4.39.0)
Installing FBSDKLoginKit (4.39.0)
Installing Firebase (5.14.0)
Installing FirebaseAuth (5.1.0)
Installing FirebaseAuthInterop (1.0.0)
Installing FirebaseCore (5.1.9)
Installing FirebaseDatabase (5.0.4)
Installing FirebaseFirestore (0.16.0)
Installing FirebaseStorage (3.0.3)
Installing FirebaseUI (5.2.2)
Installing GTMSessionFetcher (1.2.1)
Installing GoogleSignIn (4.4.0)
Installing GoogleToolboxForMac (2.2.0)
Installing GoogleUtilities (5.3.6)
Installing Protobuf (3.6.1)
Installing SDWebImage (4.4.3)
Installing TwitterCore (3.2.0)
Installing TwitterKit (3.4.2)
Installing gRPC-C++ (0.0.5)
Installing gRPC-Core (1.14.0)
Installing leveldb-library (1.20)
Installing nanopb (0.3.901)
Generating Pods project
```

As you can see, it fetches FBSDKCoreKit 4.39.0 even though in the FirebaseUI Podfile it states that the version should be 4.35.0.

I only noticed it now because 4.39.0 has a bug which broke the FB login functionality. However, I am also worried that in the future I might run into similar issues.

Am I doing something wrong?

My `pod --version` is 1.5.3

**Comment by mwawrusch** [comment]
Thanks for pointing this out - just wondering why I can't login anymore...

**Comment by AndriyGo** [comment]
@mwawrusch yes, it is really frustrating. Especially the fact that the guys at FB still haven't submitted a new release which fixes the issue even though they already have a commit which addresses it in their repo! So, we are stuck with Firebase pod not processing correctly and fetching the latest version of FBSDK instead of 4.35.0 + FB not releasing an urgent patch :(

**Comment by morganchen12** [comment]
Sorry about this all. Restricting the dependency version of the Facebook SDK in our podspec is technically a breaking change, so we haven't done so yet. Our next change is a major version bump, though, so this fix will be released with 6.0.

**Comment by AndriyGo** [comment]
@morganchen12 thanks for the update. 

1) ```target 'FirebaseFacebookAuthUI' do
    inherit! :search_paths
    # Pods for Facebook Auth
    # These are pinned to 4.35.0 to work around this bug:
    # https://developers.facebook.com/support/bugs/242258916492125/?disable_redirect=0
    pod 'FBSDKLoginKit', '= 4.35.0'
    pod 'FBSDKCoreKit', '= 4.35.0'```

I thought this was included to restrict the version of FBSDK to 4.35.0?

2) Do you know approximate timeframe when 6.0 will be released\\\\\\\/

**Comment by morganchen12** [comment]
It's in our Podfile so that this project doesn't break, but it's not in our Podspec so the change isn't propagated to projects depending on FirebaseUI. I'm aiming to release FirebaseUI 6 by the end of the month.

**Comment by morganchen12** [comment]
You can pin the version in your own Podfile by copy-pasting the same lines from the Podfile in this project.

**Comment by AndriyGo** [comment]
@morganchen12 thank you for the suggestion, it works!

**Comment by ruggierom** [comment]
This helped me too. Thanks!

**Comment by morganchen12** [comment]
Hey all, the release didn't happen at end-of-month as planned. It'll take at least another week--some of our testing infrastructure broke that I haven't gotten around to fixing. Sorry for the delay.

**Comment by kmurphy002** [comment]
This issue has seemingly reappeared. I installed the pod FirebaseUI/Facebook which pulls in the firebaseloginkit 4.4.1 (I'm recalling the version from memory since I am at work and my project is at home); after which I opened the project via the firebase workspace file, and I get "arc semantic" errors in one of the facebook header files.

So this indicated to me that something was completely broken. After reading this thread I figured I try and manually install the facebookloginkit pod and remove the FirebaseUI/Facebook pod. The version for facebookloginkit was now 5.0 and my errors in the library headers are now gone.

any workaround for this if you still want to leverage the FirebaseUI pod? Or any updates if there is a current fix for this would be great? Or if I am taking crazy pills and somehow this is user error, let me know.
